src/main/java/com/example/todolist/Domain/Entity/User.java

```diff
@@ -0,0 +1,77 @@
+package com.example.todolist.Domain.Entity;
+
+import com.example.todolist.Domain.Validation.UserValidation.EmailRule;
+import com.example.todolist.Domain.Validation.UserValidation.PasswordRule;
+import com.example.todolist.Domain.Validation.UserValidation.PhoneRule;
+import com.example.todolist.Domain.Validation.UserValidation.UsernameRule;
+import jakarta.persistence.*;
+
+import java.util.List;
+
+
+@Entity
+@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
+public class User extends BaseEntity {
+    @Column(unique = true, nullable = false)
+    @UsernameRule
+    private String username;
+    @Column(nullable = false)
+    @PasswordRule
+    private String password;
+    @Column(nullable = false, unique = true)
+    @EmailRule
+    private String email;
+    @PhoneRule
+    private String phone;
+
+    @OneToMany(mappedBy = "user")
+    private List<Task> tasks;
+
+    public User() {}
```

Comment:
mark this as protected not public, since this is not a business logic, every new user has to be passed to an information, we have this constructor because this is required by jakatar